### PR TITLE
Add nullable support to Swift4

### PR DESCRIPTION
**Pull request by wing328**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Add `nullable` (or x-nullable in OAS2) support to swift4

This will replace the unwrapRequired option.

Tests passed locally:
```
2019-08-28 18:24:13.135 xcodebuild[39128:3957771] [MT] IDETestOperationsObserverDebug: (2DC50469-4B44-406B-9B42-2EE085D7192E) Finished requesting crash reports. Continuing with testing.
All tests
Test Suite SwaggerClientTests.xctest started
PetAPITests
    ✓ test1CreatePet (0.697 seconds)
    ✓ test2GetPet (0.505 seconds)
    ✓ test3DeletePet (0.497 seconds)
StoreAPITests
    ✓ test1PlaceOrder (0.723 seconds)
    ✓ test2GetOrder (0.479 seconds)
    ✓ test3DeleteOrder (0.483 seconds)
UserAPITests
    ✓ testLogin (0.450 seconds)
    ✓ testLogout (0.530 seconds)


Executed 8 tests, with 0 failures (0 unexpected) in 4.362 (4.366) seconds
2019-08-28 18:27:13.958 xcodebuild[39128:3957771] [MT] IDETestOperationsObserverDebug: 205.584 elapsed -- Testing started completed.
2019-08-28 18:27:13.958 xcodebuild[39128:3957771] [MT] IDETestOperationsObserverDebug: 0.000 sec, +0.000 sec -- start
2019-08-28 18:27:13.958 xcodebuild[39128:3957771] [MT] IDETestOperationsObserverDebug: 205.584 sec, +205.584 sec -- end
▸ Test Succeeded
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 04:55 min
[INFO] Finished at: 2019-08-28T18:27:20+08:00
[INFO] ------------------------------------------------------------------------
```

cc @jgavris (2017/07) @ehyche (2017/08) @Edubits (2017/09) @jaz-ah (2017/09) @d-date (2018/03)

### CODE REVIEW

**Comment by d-date** [comment]
@wing328 please install swiftlint via homebrew

**Comment by wing328** [comment]
@d-date ok. I'll do it tomorrow.

But overall do you like using nullable to replace the unwrapRequired option?

**Comment by d-date** [comment]
@wing328 I cannot imagine how `nullable` works. Is it completely equal to unwrapRequired?
I need to check spec.

**Comment by wing328** [comment]
👌 
```
nullabe: Allows sending a null value for the defined schema. Default value is false.
```

Ref: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#schemaNullable

**Comment by wing328** [comment]
> @wing328 please install swiftlint via homebrew

Done and updated the samples:

```
➜  openapi-generator git:(swift_nullable_support) swiftlint version
0.34.0
```

File: samples/client/petstore/swift4/unwrapRequired/PetstoreClient/Classes/OpenAPIs/Models/TypeHolderExample.swift [hunk 1]
```diff
@@ -9,11 +9,11 @@ import Foundation
...
-    public var stringItem: String?
-    public var numberItem: Double?
-    public var integerItem: Int?
-    public var boolItem: Bool?
-    public var arrayItem: [Int]?
+    public var stringItem: String
+    public var numberItem: Double
+    public var integerItem: Int
+    public var boolItem: Bool
+    public var arrayItem: [Int]
```
Comment (d-date) [inline samples/client/petstore/swift4/unwrapRequired/PetstoreClient/Classes/OpenAPIs/Models/TypeHolderExample.swift]: This part should be Optional

File: samples/client/petstore/swift4/unwrapRequired/PetstoreClient/Classes/OpenAPIs/Models/TypeHolderDefault.swift [hunk 2]
```diff
@@ -9,11 +9,11 @@ import Foundation
...
-    public var stringItem: String? = "what"
-    public var numberItem: Double?
-    public var integerItem: Int?
-    public var boolItem: Bool? = true
-    public var arrayItem: [Int]?
+    public var stringItem: String = "what"
+    public var numberItem: Double
+    public var integerItem: Int
+    public var boolItem: Bool = true
+    public var arrayItem: [Int]
```
Comment (d-date) [inline samples/client/petstore/swift4/unwrapRequired/PetstoreClient/Classes/OpenAPIs/Models/TypeHolderDefault.swift]: This part should be Optional

File: samples/client/petstore/swift4/unwrapRequired/PetstoreClient/Classes/OpenAPIs/Models/Pet.swift [hunk 3]
```diff
@@ -16,8 +16,8 @@ public struct Pet: Codable {
     }
     public var id: Int64?
     public var category: Category?
-    public var name: String?
-    public var photoUrls: [String]?
+    public var name: String
+    public var photoUrls: [String]
```
Comment (d-date) [inline samples/client/petstore/swift4/unwrapRequired/PetstoreClient/Classes/OpenAPIs/Models/Pet.swift]: This part should be Optional

File: samples/client/petstore/swift4/unwrapRequired/PetstoreClient/Classes/OpenAPIs/Models/Name.swift [hunk 4]
```diff
@@ -11,7 +11,7 @@ import Foundation
 
 public struct Name: Codable {
 
-    public var name: Int?
+    public var name: Int
```
Comment (d-date) [inline samples/client/petstore/swift4/unwrapRequired/PetstoreClient/Classes/OpenAPIs/Models/Name.swift]: This part should be Optional

File: samples/client/petstore/swift4/unwrapRequired/PetstoreClient/Classes/OpenAPIs/Models/Category.swift [hunk 5]
```diff
@@ -10,7 +10,7 @@ import Foundation
 public struct Category: Codable {
 
     public var id: Int64?
-    public var name: String? = "default-name"
+    public var name: String = "default-name"
```
Comment (d-date) [inline samples/client/petstore/swift4/unwrapRequired/PetstoreClient/Classes/OpenAPIs/Models/Category.swift]: This part should be Optional

File: samples/client/petstore/swift4/unwrapRequired/PetstoreClient/Classes/OpenAPIs/Models/Cat.swift [hunk 6]
```diff
@@ -9,7 +9,7 @@ import Foundation
 
 public struct Cat: Codable {
 
-    public var className: String?
+    public var className: String
```
Comment (d-date) [inline samples/client/petstore/swift4/unwrapRequired/PetstoreClient/Classes/OpenAPIs/Models/Cat.swift]: This part should be Optional

File: samples/client/petstore/swift4/unwrapRequired/PetstoreClient/Classes/OpenAPIs/Models/Animal.swift [hunk 7]
```diff
@@ -9,7 +9,7 @@ import Foundation
 
 public struct Animal: Codable {
 
-    public var className: String?
+    public var className: String
```
Comment (d-date) [inline samples/client/petstore/swift4/unwrapRequired/PetstoreClient/Classes/OpenAPIs/Models/Animal.swift]: This part should be Optional

File: samples/client/petstore/swift4/unwrapRequired/PetstoreClient/Classes/OpenAPIs/Models/Dog.swift [hunk 8]
```diff
@@ -9,7 +9,7 @@ import Foundation
 
 public struct Dog: Codable {
 
-    public var className: String?
+    public var className: String
```
Comment (d-date) [inline samples/client/petstore/swift4/unwrapRequired/PetstoreClient/Classes/OpenAPIs/Models/Dog.swift]: This part should be Optional

File: samples/client/petstore/swift4/unwrapRequired/PetstoreClient/Classes/OpenAPIs/Models/FormatTest.swift [hunk 9]
```diff
@@ -12,16 +12,16 @@ public struct FormatTest: Codable {
     public var integer: Int?
     public var int32: Int?
     public var int64: Int64?
-    public var number: Double?
+    public var number: Double
```
Comment (d-date) [inline samples/client/petstore/swift4/unwrapRequired/PetstoreClient/Classes/OpenAPIs/Models/FormatTest.swift]: This part should be Optional

File: samples/client/petstore/swift4/unwrapRequired/PetstoreClient/Classes/OpenAPIs/Models/FormatTest.swift [hunk 10]
```diff
@@ -12,16 +12,16 @@ public struct FormatTest: Codable {
...
     public var float: Float?
     public var double: Double?
     public var string: String?
-    public var byte: Data?
+    public var byte: Data
```
Comment (d-date) [inline samples/client/petstore/swift4/unwrapRequired/PetstoreClient/Classes/OpenAPIs/Models/FormatTest.swift]: This part should be Optional

File: samples/client/petstore/swift4/unwrapRequired/PetstoreClient/Classes/OpenAPIs/Models/FormatTest.swift [hunk 11]
```diff
@@ -12,16 +12,16 @@ public struct FormatTest: Codable {
...
     public var binary: URL?
-    public var date: Date?
+    public var date: Date
```
Comment (d-date) [inline samples/client/petstore/swift4/unwrapRequired/PetstoreClient/Classes/OpenAPIs/Models/FormatTest.swift]: This part should be Optional

File: samples/client/petstore/swift4/unwrapRequired/PetstoreClient/Classes/OpenAPIs/Models/FormatTest.swift [hunk 12]
```diff
@@ -12,16 +12,16 @@ public struct FormatTest: Codable {
...
     public var dateTime: Date?
     public var uuid: UUID?
-    public var password: String?
+    public var password: String
```
Comment (d-date) [inline samples/client/petstore/swift4/unwrapRequired/PetstoreClient/Classes/OpenAPIs/Models/FormatTest.swift]: This part should be Optional

**Comment by wing328** [comment]
Thanks for the review. Let me work on these items this weekend...

**Comment by fl034** [comment]
Any news on this?

**Comment by wing328** [comment]
Closing this one and use https://github.com/OpenAPITools/openapi-generator/pull/4825 instead.

**Pull request closed without merging**